Userland/Libraries/LibChess/UCIEndpoint.cpp

```diff
@@ -53,15 +53,30 @@ void Endpoint::event(Core::Event& event)
         return handle_bestmove(static_cast<BestMoveCommand const&>(event));
     case Command::Type::Info:
         return handle_info(static_cast<InfoCommand const&>(event));
+    case Command::Type::Quit:
+        return handle_quit();
     default:
+        Object::event(event);
         break;
     }
 }
 
+void Endpoint::custom_event(Core::CustomEvent& custom_event)
+{
+    if (custom_event.custom_type() == EndpointEventType::UnexpectedEof)
+        handle_unexpected_eof();
+}
+
 void Endpoint::set_in_notifier()
 {
     m_in_notifier = Core::Notifier::construct(m_in->fd(), Core::Notifier::Read);
     m_in_notifier->on_ready_to_read = [this] {
+        if (m_in->eof()) {
```

Comment:
Should we only test EOF? Isn't `!m_in->can_read_line()` a better condition?

Reply:
Makes sense. I've changed it.